tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -0,0 +1,365 @@
+	BT_MESH_MODEL_CFG_SRV,
+	BT_MESH_MODEL_CFG_CLI(&cfg_cli_tx),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_1, model_op_1, &model_1_pub, NULL, &test_model_1_cb),
+};
+
+static struct bt_mesh_elem tx_elems[] = {
+	BT_MESH_ELEM(0, tx_models, BT_MESH_MODEL_NONE),
+};
+
+static const struct bt_mesh_comp tx_comp = {
+	.cid = TEST_VND_COMPANY_ID,
+	.vid = 0xeeee,
+	.pid = 0xaaaa,
+	.elem = tx_elems,
+	.elem_count = ARRAY_SIZE(tx_elems),
+};
+
+static struct bt_mesh_cfg_cli cfg_cli_rx;
+static struct bt_mesh_model rx_models[] = {
+	BT_MESH_MODEL_CFG_SRV,
+	BT_MESH_MODEL_CFG_CLI(&cfg_cli_rx),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_2, model_op_2, NULL, NULL, &test_model_2_cb),
+};
+
+static struct bt_mesh_elem rx_elems[] = {
+	BT_MESH_ELEM(0, rx_models, BT_MESH_MODEL_NONE),
+};
+
+static const struct bt_mesh_comp rx_comp = {
+	.cid = TEST_VND_COMPANY_ID,
+	.vid = 0xbaaa,
+	.pid = 0xb000,
+	.elem = rx_elems,
+	.elem_count = ARRAY_SIZE(rx_elems),
+};
+
+static int model_1_init(const struct bt_mesh_model *model)
+{
+	bt_mesh_model_msg_init(model->pub->msg, TEST_MESSAGE_OP);
+	net_buf_simple_add_u8(model->pub->msg, 1);
+
+	return 0;
+}
+
+static int model_2_init(const struct bt_mesh_model *model)
+{
+	return 0;
+}
+
+static void provision_and_configure(struct bt_mesh_test_cfg cfg, bool rx)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, cfg.addr, cfg.dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+	}
+
+	err = bt_mesh_cfg_cli_app_key_add(0, cfg.addr, 0, 0, app_key, &status);
+	if (err || status) {
+		FAIL("AppKey add failed (err %d, status %u)", err, status);
+	}
+
+	if (rx) {
+		err = bt_mesh_cfg_cli_mod_app_bind(0, rx_cfg.addr, rx_cfg.addr, 0, TEST_MODEL_ID_2,
+						   &status);
+		if (err || status) {
+			FAIL("Model %#4x bind failed (err %d, status %u)", TEST_MODEL_ID_2, err,
+			     status);
+		}
+	} else {
+		err = bt_mesh_cfg_cli_mod_app_bind(0, tx_cfg.addr, tx_cfg.addr, 0, TEST_MODEL_ID_1,
+						   &status);
+		if (err || status) {
+			FAIL("Model %#4x bind failed (err %d, status %u)", TEST_MODEL_ID_1, err,
+			     status);
+		}
+	}
+}
+
+struct bt_mesh_cfg_cli_mod_pub pub_params = {
+	.addr = rx_cfg.addr,
+	.uuid = NULL,
+	.cred_flag = false,
+	.app_idx = 0,
+	.ttl = 5,
+	.period = BT_MESH_PUB_PERIOD_SEC(1),
+	.transmit = 0,
+};
+
+static void test_tx_suspend_resume(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &tx_comp);
+	provision_and_configure(tx_cfg, 0);
+
+	k_sem_init(&publish_sem, 0, 1);
+	suspended = false;
+	uint8_t status;
+	int err;
+
+	err = bt_mesh_cfg_cli_mod_pub_set(0, tx_cfg.addr, tx_cfg.addr, TEST_MODEL_ID_1, &pub_params,
+					  &status);
+	if (err || status) {
+		FAIL("Mod pub set failed (err %d, status %u)", err, status);
+	}
+
+	/* Wait until node has published before suspending. */
+	for (int i = 0; i < NUM_PUB; i++) {
+		err = k_sem_take(&publish_sem, K_SECONDS(30));
+		if (err) {
+			FAIL("Pub timed out");
+		}
+	}
+
+	err = bt_mesh_suspend();
```

Comment:
Please use in all such places `ASSERT_OK` macro. Places where just check error code. There are many in the file. 
```
ASSERT_OK(bt_mesh_suspend());
```
It will add automatically general message with error code if it happened.
If you need additional message, just use `ASSERT_OK_MSG`
```
for (int i = 0; i < NUM_PUB; i++) {
	ASSERT_OK_MSG(k_sem_take(&publish_sem, K_SECONDS(30)), "Pub timed out");
}
```

Reply:
Fixed, thanks for the suggestion!